### ISSUE TITLE
Unrar

### ISSUE THREAD

**Issue by DemonCow5761** [Radarr/Radarr#1771]
I guess I just wanted to say I've really enjoyed using radarr (and sonarr) since I got it a couple months ago. I only use torrents and all the private trackers I'm apart of do use .rar for most tv and movies. The deluge extractor has been broken for a long time and only works sometimes and even if it did work I guess it wouldn't delete the unrard file after being copied by sonarr anyways. I'm on Windows 10 and I've spent around 8 hours straight for the last two nights trying to automate the unrar process and also would like the unrard video file to be deleted after being copied by sonarr and haven't been able to do it. I don't really know how to code n write scripts and things that advanced. I mean I can run sonarr radarr jackett with deluge n plex n share with my friends so I'm not like retarded.

I just wish there was a nice option inside radarr (and sonarr) where I just enable unrar and it unrars and deletes the extra video file. I just don't think it should be this hard and it seems like it's easier for you guys to do something than the torrent clients unraring and some how deleting the extra file after its copied.

Anyways I've given up for now unless I can get seanvree's method (edit: got it working) (https://forums.sonarr.tv/t/guide-automatically-process-rar-releases-in-sonarr-radarr-while-using-usenet-and-torrents-on-a-windows-environment/13743) working sometime and then I'll be screwed once the drone factory is gone I guess? Unless somebody can point me to step by step instructions for somebody on windows 10 to create and run these scripts because I haven't found any guides that will unrar and delete the extra video file other than seanvree's.

Kinda copypasta from a thread i made on sonarr.

TLDR 
Plz add support in radarr to unrar and delete the extra video file after being copied to sonarr.  It shouldn't be this complicated.

**Comment by galli-leo** [comment]
Sadly adding unrar support is not that easy. IMO this should be handled by your download client. It is responsible for processing your downloads. If it doesn't work in deluge try asking on their support page. Maybe you are missing something.

I can add this as a feature request, but I doubt it's going to be implemented soon, as most dl clients have functionality for that just fine.

**Comment by DemonCow5761** [comment]
Yea I don't think I'm missing anything.  I know people can make execute scripts and such but I am not skilled enough to do so, and could find no guides anywhere except some linux stuff.  As for Deluge, they have to know Extractor plugin is broken there are a lot of threads about it already, I don't know if they just can't fix it or what, basically it only works sometimes.  Just sucks we have to even use blackhole, drone folder and extractnow instead of just using Deluge download client option in radarr with an auto-unrar option.  Also sonarr is removing drone factory so I'll have to run sonnar differently than radarr eventually which sucks for newbs like me who would like to just do the same setup exactly twice.

Can you recommend any torrent dl clients with working unrar functionality in windows 10?  I haven't seen anything in qbit or utorrent and that's all I've ever tried besides Deluge.  Even if the client can unrar can the client auto delete it with the torrent data after seeding it?  Doesn't it just remain with the folder until manually deleted, correct?  Because even when the extractor did work I had to go in and manually delete the video file after as deluge doesn't recognize it as part of the torrent data.  So seems like the client cannot fully automate the unrar and delete process without vast knowledge of creating and using scripts.

Thanks for adding it as a feature request, I don't expect anything to happen anytime soon or anything I just wanted to make sure you guys know that it is quite a headache for a lot of us.  Hope you can understand what I'm trying to say I'm really tired n loopy right now.

**Comment by hotio** [comment]
You can use Deluge just fine with CDH (no drone), as a downloader(no blackhole) and it can unpack just fine using a simple script.

**Comment by DemonCow5761** [comment]
Yea I only have the one rig that I use for gaming as well so linux isn't really an option without spending money and atm I cannot.

**Comment by theCheek** [comment]
@DemonCow5761 have you tried https://github.com/cvarta/deluge-extractor ?

It's an improvement over the broken extractor plugin of deluge, and works just fine for me.

Now you just need to find a batch script to be triggerred by Radarrr to check if there are rar files present, and if so, delete the video file. 

I would avoid following the instructions of seanvree's method, as it involves the drone factory which is deprecated - no point implementing something without future proofing right?

**Comment by levic92** [comment]
I have made my own plugin to better support Sonarr and Radarr, feel free to try it out. It has been working great for me. It doesn't clean up the files but this may be something I will look at adding in the future.

https://github.com/levic92/LCExtractor

**Comment by rubdos** [comment]
I'm not sure whether it's better to open a new issue about this, or add it here:

Kodi plays rar files natively. It'd be interesting to have radarr handle .rar files the same as any movie format. That way, one doesn't lose space due to having to extract the files.

In short: choosing between extracting and "keep rars as is" would be nice. Just linking them would be fine in this usecase.

**Comment by levic92** [comment]
Here is a script I created to clean up extracted files from torrents downloaded in deluge [https://github.com/levic92/LCExtractor/tree/master/extras](url)

**Comment by theCheek** [comment]
@levic92 just implemented your clean up script. Works perfect. This is the final piece of the puzzle. Many thanks!

**Comment by becian** [comment]
@levic92 
i followed the instructions on my synology box, python 2/3 installed, modified the .py file with the ip address/password. put the script in a shared directory and followed that setup on the extras page but im getting this:

> lcextractor_cleanup.py: No such file or directory
> | lcextractor_cleanup.py /usr/bin/env: python

any idea whats going on here?

**Comment by DemonCow5761** [comment]
This is still a problem for me and many others.  There is no way to auto extract on windows reliably except using things like extract now which leave tons of left over files(can use lcextractor cleanup with it, but doesn't fix the next problem) and re-extracts everything if your computer restarts.  No extraction plug-in for deluge works all the time, same problem with qbit.  Transmission might work but most sites don't allow that.  Best I've found is LCExtractor with LCEtractor Cleanup but I would say the successful extraction rate is somewhere around 50%.

If anybody has a solution that would be great, it's been years... the original request on sonarr is from 2015!  Don't see how this isn't added couchpotato had it years ago.

> as most dl clients have functionality for that just fine.

If your still around galli-leo I would love to know these... as I haven't found a single one that works properly on windows 10 in 2 years.

TLDR: Unrar still needed.  Plz help us torrent users on windows, thanks.

**Comment by 0xdevalias** [comment]
Just wanted to link to a little codebase digging I did on Sonarr (though likely also still applicable here) with regards to what/where might be required to implement unrar support (initially targetting the 'manual import' type feature) in case it gives people ideas on where to start looking to implement this: * https://github.com/Sonarr/Sonarr/issues/784#issuecomment-478204150

**Comment by bakerboy448** [comment]
No plans to have radarr do this.

various torrent clients can handle this.

**Comment by rubdos** [comment]
I'll leave this here, for future visitors: https://github.com/davidnewhall/unpackerr

This application runs as a daemon on your download host. It checks for completed downloads and extracts them so Radarr and/or Sonarr and/or Lidarr may import them.

**Comment by DemonCow5761** [comment]
> I'll leave this here, for future visitors: https://github.com/davidnewhall/unpackerr
> 
> This application runs as a daemon on your download host. It checks for completed downloads and extracts them so Radarr and/or Sonarr and/or Lidarr may import them.

This is what I've started using as well it also deletes the leftover unrar'd files for you after sonarr/radarr imports and works every time unlike other solutions that are hit or miss for me on windows 10.
